rasa_sdk/forms.py

```diff
@@ -817,7 +817,12 @@ def slots_mapped_in_domain(self, domain: "DomainDict") -> List[Text]:
         Returns:
             Slot names mapped in the domain.
         """
-        return list(domain.get("forms", {}).get(self.form_name(), {}).keys())
+        form = domain.get("forms", {}).get(self.form_name(), {})
+        if "required_slots" in form:
```

Comment:
let's add a comment why we have to do this to the code to make it clearer for developers which read the code in the future